source/patterns/@aws-solutions-constructs/core/lib/elasticsearch-helper.ts

```diff
@@ -228,48 +227,63 @@ export function buildElasticSearchCWAlarms(scope: Construct): cloudwatch.Alarm[]
   return alarms;
 }
 
-function GetSubnetsAndSecurityGroups(vpc?: ec2.IVpc, props?: elasticsearch.CfnDomainProps, lambdaFunction?: lambda.Function):
+function getSubnetIds(vpc?: ec2.IVpc, props?: elasticsearch.CfnDomainProps):
   elasticsearch.CfnDomainProps {
   if (vpc) {
     // Environment specified stacks: A ES cluster deploys in 3 AZs with 3 subnets maximum(each subnet in a different AZ).
-    // Environment-agnostic stacks: A ES cluster deploys in 2 AZs with 2 subnets maximum(each subnet in a different AZ).
+    // Environment agnostic stacks: A ES cluster deploys in 2 AZs with 2 subnets maximum(each subnet in a different AZ).
     // To successfully deploy a ES cluster, construct will need to specify exactly same subnets number as cluster AZs number
     // To summarize: 1. Subnets number must equals to AZs number.   2. Each subnet belongs to different AZ.
     let subnetIds: string[] = vpc.selectSubnets({ onePerAz: true }).subnetIds;
 
     if (props?.elasticsearchClusterConfig) {
-      const disabledZoneAwareness = 'zoneAwarenessEnabled' in props.elasticsearchClusterConfig &&
-        !props.elasticsearchClusterConfig.zoneAwarenessEnabled;
-      const twoAvailabilityZoneConfig = 'zoneAwarenessConfig' in props.elasticsearchClusterConfig &&
-        props.elasticsearchClusterConfig.zoneAwarenessConfig &&
-        'availabilityZoneCount' in props.elasticsearchClusterConfig.zoneAwarenessConfig &&
-        props.elasticsearchClusterConfig.zoneAwarenessConfig.availabilityZoneCount === 2;
+      const disabledZoneAwareness = checkDisabledZoneAwareness(props);
+      const twoAvailabilityZoneConfig = checkTwoAvailabilityZoneConfig(props);
 
-      if (disabledZoneAwareness) { // If zoneAwarenessEnabled set to false, ES cluster deploy in 1 AZ with 1 subnet
-        subnetIds = subnetIds.slice(0, 1);
-      } else if (twoAvailabilityZoneConfig) { // If zoneAwarenessConfig.availabilityZoneCount set to 2, ES cluster deploy in 2 AZ with 2 subnets.
-        subnetIds = subnetIds.slice(0, 2);
+      if (disabledZoneAwareness) { // If zone awareness is disabled, ES cluster deploy in 1 AZ with 1 subnet
+        subnetIds = extractSubnetIdsByLength(subnetIds, 1);
```

Comment:
extractSubnetIdsByLength

Huh? Length of what? Aside from a change about 6 years ago, aren't all Ids the same length?